chain/chain/src/chain.rs

```diff
@@ -2734,9 +2733,24 @@ impl<'a> ChainUpdate<'a> {
             Some(&block.hash()),
         )?;
         self.chain_store_update.save_block_extra(&block.hash(), BlockExtra { challenges_result });
+        let prev_protocol_version =
+            self.runtime_adapter.get_epoch_protocol_version(prev_block.header().epoch_id())?;
         let protocol_version =
             self.runtime_adapter.get_epoch_protocol_version(block.header().epoch_id())?;
 
+        // Re-introduce receipts missing before apply_chunks fix (see https://github.com/near/nearcore/pull/4228)
+        #[cfg(feature = "protocol_feature_restore_receipts_after_fix")]
+        if self.runtime_adapter.is_next_block_epoch_start(prev_block.hash()).unwrap_or(false)
+            && prev_protocol_version < ProtocolFeature::RestoreReceiptsAfterFix.protocol_version()
+            && ProtocolFeature::RestoreReceiptsAfterFix.protocol_version() <= protocol_version
+        {
+            let receipt_result_json =
+                include_str!("../../../neard/res/fix_apply_chunks_receipts.json");
+            let receipt_result = serde_json::from_str::<ReceiptResult>(receipt_result_json)
```

Comment:
Seems like you apply receipts saved from mainnet on all nets, this may lead to weird results